source/core/ut_suite_manager.pkb

```diff
@@ -327,5 +339,166 @@ create or replace package body ut_suite_manager is
+        
+          l_object_name := l_item.object_name;
+          --l_object_name := regexp_substr(l_object_name,'\w+$'); -- temporary fix. seems like suite have suitepath in object_name
+          if regexp_like(l_object_name, l_root, modifier => 'i') then
+            
+            set_skipped_flag(l_item, l_rest_path);          
+            l_items.extend;
+            l_items(l_items.count) := l_item;
+
+          end if;
+          
+        end loop;
+        
+        a_suite.items := l_items;
+      
+        if l_items.count = 0 then--not l_found then
+          raise_application_error(-20203, 'Suite note found');
+        end if;
+      end if;
+    end set_skipped_flag;
+  begin
+    l_paths := clean_paths(a_paths);
+  
+    validate_paths(l_paths);
+  
+    -- current implementation operates only on a single path
+    -- to be improved later
+    for i in 1 .. l_paths.count loop
+      l_path   := l_paths(i);
+      l_schema := regexp_substr(l_path, '^(\w+)(\.|:|$)', 1, 1, null, 1);
+    
+      l_schema_suites := get_schema_suites(upper(l_schema));
+    
+      if regexp_like(l_path, '^\w+$') then
+        -- run whole schema
+        l_index := l_schema_suites.first;
+        while l_index is not null loop
+          l_objects_to_run.extend;
+          l_objects_to_run(l_objects_to_run.count) := l_schema_suites(l_index);
+          l_index := l_schema_suites.next(l_index);
+        end loop;
+      else
+        -- convert SCHEMA.PACKAGE.PROCEDURE syntax to fully qualified path
+        if regexp_like(l_path, '^\w+(\.\w+){1,2}$') then
+          declare
+            l_temp_suite     ut_test_suite;
+            l_package_name   varchar2(4000);
+            l_procedure_name varchar2(4000);
+          begin
+            l_package_name   := regexp_substr(l_path, '^\w+\.(\w+)(\.(\w+))?$', subexpression => 1);
+            l_procedure_name := regexp_substr(l_path, '^\w+\.(\w+)(\.(\w+))?$', subexpression => 3);
+          
+            l_temp_suite := config_package(l_schema, l_package_name);
+            l_path       := rtrim(l_schema || ':' || l_temp_suite.object_path || '.' || l_procedure_name, '.');
+          end;
+        end if;
+        
+        if regexp_like(l_path, '^\w+:.+$') then
+          -- fully quilified path branch in the form
+          -- schema:suite.suite.suite
+        
+          l_suite_path      := regexp_substr(l_path, ':(.+)', subexpression => 1);
+          l_root_suite_name := regexp_substr(l_suite_path, '^\w+');
+        
+          l_suite := l_schema_suites(l_root_suite_name);
+        
+          --set_skipped_flag(l_suite, l_suite_path);
+          set_skipped_flag(l_suite, regexp_substr(l_suite_path, '\.(.+)', subexpression => 1));
+        
+          l_objects_to_run.extend;
+          l_objects_to_run(l_objects_to_run.count) := l_suite;
+        end if;
+
+      end if;
+    
+    end loop;
+  
+    if l_objects_to_run.count > 0 then
+      l_reporter.before_run(a_suites => l_objects_to_run);
```

Comment:
Nice.
Now we pass the full list of objects to execute to the before/after calls.